### ISSUE TITLE
Game not using monitor resolution

### ISSUE THREAD

**Issue by licaon-kter** [virtual-programming/specops-linux#3]
As per Witcher2, still it can't use nVidia downsampling ( say scale 3840x2400 on a 1920x1200 display ) due to SDL(!?): https://github.com/virtual-programming/witcher2-linux/issues/35#issuecomment-53136674

SO:TL https://i.imgur.com/oQ3NY5O.jpg ( observe how the steam FPS overlay behaves like it should but the game does not )
vs
Pillars of Eternity: https://i.imgur.com/6PJBlxQ.jpg

**Comment by vpltd-peterm** [comment]
We're working on replacing all our SDL2 code with native X11 code, so we can control exactly what is going on. We don't have a time frame on this yet, but when complete we will likely update all our games

**Comment by Tux1c** [comment]
@vpltd-peterm is using barebones X11 code really a good solution?

What happens when DEs get fully ported over to Wayland? What happens under Ubuntu with Mir?
I mean, sure, you could use some X11 compatability layer, but AFAIK, the xWayland (or however the layer is called) is very sluggish and works poorly, and if I'm not mistaken 3D graphics are not supported.
If it was an old game we're talking about, then sure, using a compatability layer is OK, but I don't think making a new game "x11 exclusive" is a good idea.

**Comment by vpltd-peterm** [comment]
This is quite true, but we are left with no alternative I think. SDL2 simply doesnt give us some feature set we need. We'll be sure to structure our code such that we could add a Wayland backend in future. I also think that XWayland's performance will improve given time.

**Comment by licaon-kter** [comment]
@Tux1c: To be fair, could you point a distro that ships Wayland as default in the next 6 months? 1 year?
Yeah, I want it NOW too, but in the mean time...

**Comment by vpltd-peterm** [comment]
Wayland will only be worth supporting from a gaming point of view, when the nvidia/amd closed source drivers support it well too.. especially the nvidia driver.

**Comment by Tux1c** [comment]
@vpltd-peterm Yeah, just wanted to make sure you guys are aware about Wayland and the whole X11-Wayland situation. I am aware of the driver issues, and I've meant only to, again, raise a flag, that you should atleast write the change so it would be easy to support Wayland in the future or something similar :)

@licaon-kter Well to be honest, I COULD give you some distros that ship with Wayland by default, or those that ship with a choice between the too (usually minimalistic distros) but that is not the point, I am aware that Wayland is immature and not so popular, I mean, KDE just started their "beta" testing on the latest, yet to be full, KDE 5.
GNOME also tries to push for Wayland, and other DEs don't even support it.
But I think it's a good practice to support, or atleast, bear in mind that a change is coming in our direction, you don't have to wait for Wayland to be very popular in order to support it, that would create an endless circle, nobody uses Wayland cuz it doesn't support their software, and their software doesn't support Wayland because nobody uses it ;)

**Comment by vpltd-peterm** [comment]
Realistically, Wayland is going to have to have good X11 support in its early days, or it will never take off... rewrites will not happen overnight!

**Comment by rafaelfdr** [comment]
Instead of removing SDL2, did you try checking with SDL devs ?

**Comment by bertvandepoel** [comment]
This is still not fixed...
